### PR TITLE
[webview_flutter_android] Adds support for use of old hybrid composition

### PR DESCRIPTION
**Pull request by bparrishMines**
Adds support for Hybrid Composition to deal with: https://github.com/flutter/flutter/issues/104686
Also fixes: https://github.com/flutter/flutter/issues/102284

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/webview_flutter/webview_flutter_android/lib/webview_surface_android.dart [hunk 1]
```diff
@@ -17,12 +17,6 @@ import 'webview_android_widget.dart';
 /// Android [WebViewPlatform] that uses [AndroidViewSurface] to build the [WebView] widget.
 ///
 /// To use this, set [WebView.platform] to an instance of this class.
-///
-/// This implementation uses hybrid composition to render the [WebView] on
-/// Android. It solves multiple issues related to accessibility and interaction
-/// with the [WebView] at the cost of some performance on Android versions below
-/// 10. See https://github.com/flutter/flutter/wiki/Hybrid-Composition for more
-/// information.
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_android/lib/webview_surface_android.dart:25]: This is no longer true for this `AndroidView`. I moved it to the new one.

File: packages/webview_flutter/webview_flutter_android/lib/webview_surface_android.dart [hunk 2]
```diff
@@ -59,7 +53,8 @@ class SurfaceAndroidWebView extends AndroidWebView {
               // WebView content is not affected by the Android view's layout direction,
               // we explicitly set it here so that the widget doesn't require an ambient
               // directionality.
-              layoutDirection: TextDirection.rtl,
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_android/lib/webview_surface_android.dart:62]: This shouldn't be a breaking change for large majority of people. This value is ignored unless support for this is explicitly added to the `AndroidManifest.xml`. For example, it is still `ltr` in our example app.

File: packages/webview_flutter/webview_flutter_android/pubspec.yaml [hunk 3]
```diff
@@ -2,11 +2,11 @@ name: webview_flutter_android
 description: A Flutter plugin that provides a WebView widget on Android.
 repository: https://github.com/flutter/plugins/tree/main/packages/webview_flutter/webview_flutter_android
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+webview%22
-version: 2.8.14
+version: 2.9.0
 
 environment:
   sdk: ">=2.14.0 <3.0.0"
-  flutter: ">=2.8.0"
+  flutter: ">=3.0.0"
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_android/pubspec.yaml:9]: Required for `initExpensiveAndroidView`

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by bparrishMines** [comment]
@stuartmorgan This is ready for another review.

I switched the implementation to use the expensive android view when the background color is set as transparent. Since background color is a creation param, changes to it are ignored after the webview is already built.

**Review by stuartmorgan — COMMENTED** [review]
Looks like this is breaking tests on the API 26 device? But otherwise LG.

(Based on the findings I wrote up in https://github.com/flutter/flutter/issues/107313 we have other problems with webview that I hadn't realized before, but this change will still be an incremental improvement.)

**Comment by bparrishMines** [comment]
@stuartmorgan I reran the test and it passed. I believe it was just a timeout. If you believe this is good, could you approve the PR?

**Pull request merged**